dLux/apertures.py

```diff
@@ -0,0 +1,1430 @@
+        """
+        Rotate the coordinate system by a pre-specified amount,
+        `self._theta`
+
+        Parameters
+        ----------
+        coordinates : Tensor
+            A `(2, npix, npix)` representation of the coordinate 
+            system. The leading dimensions specifies the x and then 
+            the y coordinates in that order. 
+
+        Returns
+        -------
+        coordinates : Tensor
+            The rotated coordinate system. 
+        """
+        rotation_matrix = np.array([
+            [np.cos(self.theta), -np.sin(self.theta)],
+            [np.sin(self.theta), np.cos(self.theta)]])            
+        return np.apply_along_axis(np.matmul, 0, coordinates, 
+            rotation_matrix) 
+
+
+    def _shear(self : Layer, coordinates : Tensor) -> Tensor:
+        """
+        Shear the coordinate system by the inbuilt amount `self._phi`.
+
+        Parameters
+        ----------
+        coordinates : Tensor
+            A `(2, npix, npix)` representation of the coordinate 
+            system. The leading dimensions specifies the x and then 
+            the y coordinates in that order. 
+
+        Returns
+        -------
+        coordinates : Tensor
+            The sheared coordinate system. 
+        """
+        return coordinates\
+            .at[0]\
+            .set(coordinates[0] - coordinates[1] * np.tan(self.phi)) 
+
+
+    def _coordinates(self : Layer) -> Tensor:
+        """
+        Generate the transformed coordinate system for the aperture.
+
+        Returns
+        -------
+        coordinates : Tensor
+            The coordinate system in the rectilinear view, with the
+            x and y coordinates stacked above one another.
+        """
+        x_pixel_offset = self.x_offset / self.pixel_scale
+        y_pixel_offset = self.y_offset / self.pixel_scale
+        coordinates = self._shear(
+            self._rotate(
+                self._magnify(
+                    self.get_pixel_scale() * \
+                        get_pixel_positions(self.pixels, 
+                            x_pixel_offset, y_pixel_offset))))
+        return coordinates
+
+
+    def set_theta(self : Layer, theta : float) -> Layer:
+        """
+        Parameters
+        ----------
+        theta : float
+            The angle of rotation from the positive x-axis.  
+
+        Returns
+        -------
+        basis : HexagonalBasis 
+            The rotated hexagonal basis. 
+        """
+        return eqx.tree_at(lambda basis : basis.theta, self, theta)
+
+
+    def set_magnification(self : Layer, rmax : float) -> Layer:
+        """
+        Parameters
+        ----------
+        rmax : float
+            The radius of the smallest circle that can completely 
+            enclose the aperture.
+
+        Returns
+        -------
+        basis : HexagonalBasis
+            The magnified hexagonal basis.
+        """
+        return eqx.tree_at(lambda basis : basis.magnification, self, rmax)
```

Comment:
Not a basis class, its an aperture class, I would change the internals for clarity